build-zones

```diff
@@ -0,0 +1,194 @@
+        return ipv4
+
+    @cached_property
+    def ipv6(self):
+        ipv6 = ipv4_to_ipv6(self.ipv4)
+        assert is_ocf_ip(ipv6), ipv6
+        return ipv6
+
+    @cached_property
+    def ldap_attrs(self):
+        with ldap_ocf() as c:
+            c.search(
+                OCF_LDAP_HOSTS,
+                '(cn={})'.format(self.hostname),
+                attributes=ldap3.ALL_ATTRIBUTES,
+            )
+            record, = c.response
+            return record['attributes']
+
+    @cached_property
+    def last_quad(self):
+        """Return the last quad from this host's IP address."""
+        return int(self.ipv4.exploded.split('.')[-1])
+
+
+    def forward_dns_records(self):
+        """Return forward DNS records."""
+        if self.ipv4:
+            yield '{self.hostname}\tIN A\t{self.ipv4}'.format(self=self)
+        if self.ipv6:
+            yield '{self.hostname}\tIN AAAA\t{self.ipv6}'.format(self=self)
+
+    def reverse_dns_records_v4(self):
+        """Return reverse IPv4 DNS records."""
+        if self.ipv4:
+            yield '{self.last_quad} IN PTR {self.hostname}.OCF.Berkeley.EDU.'.format(self=self)
+
+    def reverse_dns_records_v6(self):
+        """Return reverse IPv6 DNS records."""
+        last_quad_dotted = '.'.join(reversed(str(self.last_quad).zfill(4)))
+        # TODO: uppercase these properly like IPv4
+        if self.ipv6:
+            yield '{} IN PTR {self.hostname}.ocf.berkeley.edu.'.format(last_quad_dotted, self=self)
+
+    def __str__(self):
+        return '{self.hostname}\tIN A\t{self.ipv4}'.format(self=self)
+
+
+def hosts_from_ldap():
+    """Return set of hosts at the OCF."""
+    with ldap_ocf() as c:
+        c.search(OCF_LDAP_HOSTS, '(cn=*)', attributes=['cn'])
+        hosts = {
+            Host(record['attributes']['cn'][0])
+            for record in c.response
+        }
+    return hosts
+
+
+def forward_file(hosts, serial):
+    return open('templates/db.ocf.tmpl').read().format(
+        banner=BANNER,
+        serial=serial,
+        records='\n\n'.join(
+            '\n'.join(
+                ['; {}'.format(host.description)] +
+                list(host.forward_dns_records())
+            )
+            for host in sorted(hosts, key=attrgetter('ipv4'))
+        )
+    )
+
+
+def reverse_file_v4(hosts, serial):
+    return open('templates/db.ocf.in-addr.arpa.tmpl').read().format(
+        banner=BANNER,
+        serial=serial,
+        records='\n'.join(chain.from_iterable(
+            host.reverse_dns_records_v4()
+            for host in sorted(hosts, key=attrgetter('ipv4'))
+        ))
+    )
+
+
+def reverse_file_v6(hosts, serial):
+    return open('templates/db.ocf.ip6.arpa.tmpl').read().format(
+        banner=BANNER,
+        serial=serial,
+        records='\n'.join(chain.from_iterable(
+            host.reverse_dns_records_v6()
+            for host in sorted(hosts, key=attrgetter('ipv6'))
+        ))
+    )
+
+
+def generate_serial():
+    # TODO: this
```

Comment:
If we're willing to break the YYYYMMDDXX format, we could also use unix time stamp plus 1000000000 or something. Guess it depends if we want it to be human-readable.

(programmers can pretty much read unix timestamps anyway, though)

Reading from the files or using dnspython to get the current serial is probably the best option. Future staff might hate us if we break YYYYMMDDXX.

If we get the current serial we can use some first-grade-level Python to increase by 1 if it's the current day, or reset to `{current_day} + '00'` otherwise